core/src/test/java/io/questdb/test/cutlass/pgwire/PGWireMetricsQueryCounterTest.java

```diff
@@ -0,0 +1,580 @@
+                Statement statement = test.statement;
+
+                boolean hasResult = statement.execute(
+                        "CREATE TABLE TEST(l long, s string); " +
+                                "INSERT INTO TEST VALUES (3, 'three'); /*some comment */" +
+                                "TRUNCATE TABLE TEST;" +
+                                "SELECT '1';");
+                assertResults(statement, hasResult, count(0), count(1),
+                        count(0), data(row("1")));
+
+                long startedQueriesForTest = metrics.pgWire().startedQueriesCounter().getValue() - totalQueriesStarted;
+                long completedQueriesForTest = metrics.pgWire().completedQueriesCounter().getValue() - totalQueriesCompleted;
+                Assert.assertEquals(6, startedQueriesForTest);
+                Assert.assertEquals(6, completedQueriesForTest);
+            }
+        });
+    }
+
+    @Test // still counts a query even though it does not have a semicolon
+    public void testQueryCountTwoBlocksEndWithoutSemicolon() throws Exception {
+        // former test name testRunCOMMITWithSemicolonReturnsNextQueryResultOnly
+        assertMemoryLeak(() -> {
+            try (PGWireMetricsQueryCounterTest.PGTestSetup test = new PGWireMetricsQueryCounterTest.PGTestSetup()) {
+                Statement statement = test.statement;
+                boolean hasResult = statement.execute("COMMIT; select 3");
+                assertResults(statement, hasResult, PGMultiStatementMessageTest.Result.ZERO, data(row(3L)));
+
+                long startedQueriesForTest = metrics.pgWire().startedQueriesCounter().getValue() - totalQueriesStarted;
+                long completedQueriesForTest = metrics.pgWire().completedQueriesCounter().getValue() - totalQueriesCompleted;
+                Assert.assertEquals(4, startedQueriesForTest);
+                Assert.assertEquals(4, completedQueriesForTest);
+            }
+        });
+    }
+
+    @Test
+    public void testQueryCountWithoutSemicolon() throws Exception {
+        // former test name testRunBEGINWithoutSemicolonReturnsNoResult
+        assertMemoryLeak(() -> {
+            try (PGWireMetricsQueryCounterTest.PGTestSetup test = new PGWireMetricsQueryCounterTest.PGTestSetup()) {
+                Statement statement = test.statement;
+                boolean hasResult = statement.execute("BEGIN");
+                assertResults(statement, hasResult, PGMultiStatementMessageTest.Result.ZERO);
+
+
+                long startedQueriesForTest = metrics.pgWire().startedQueriesCounter().getValue() - totalQueriesStarted;
+                long completedQueriesForTest = metrics.pgWire().completedQueriesCounter().getValue() - totalQueriesCompleted;
+                Assert.assertEquals(3, startedQueriesForTest);
+                Assert.assertEquals(3, completedQueriesForTest);
+            }
+        });
+    }
+
```

Comment:
It'd be better to aviod copying all this code, especially that you don't really need to assert results, only counters .